flink-sql-runner/src/main/java/com/github/streamshub/flink/KubernetesSecretInterpolator.java

```diff
@@ -24,14 +24,14 @@ public KubernetesSecretInterpolator() {
 
     public String interpolate(String input) {
         Matcher matcher = SECRET_PATTERN.matcher(input);
-        StringBuffer result = new StringBuffer();
+        StringBuilder result = new StringBuilder();
 
         while (matcher.find()) {
             String namespace = matcher.group(1);
             String secretName = matcher.group(2);
             String secretKey = matcher.group(3);
             String secretValue = getSecretValue(namespace, secretName, secretKey);
-            matcher.appendReplacement(result, secretValue != null ? secretValue.trim() : "");
+            matcher.appendReplacement(result, secretValue.trim());
```

Comment:
What happens if a secret is not found? Does it throw or return null? Should `getSecreteValue` return an `Optional<String>` so that one error doesn't stop progress replacing other secrets?

Reply:
I think the secretValue will always be not null because getSecretValue() will throw NPE when the secret is not found.